odoo/tools/translate.py

```diff
@@ -752,7 +752,7 @@ def write_rows(self, rows):
             row.setdefault('modules', set()).add(module)
             if not row.get('translation') and trad != src:
                 row['translation'] = trad
-            row.setdefault('tnrs', []).append((type, name, res_id))
+            row.setdefault('tnrs', set()).add((type, name, res_id if type != 'code' else 0))
```

Comment:
`if type != 'code' else 0` wasn't there before, so why is it needed now? Is it related to your changes?

Reply:
I'm setting `res_id = 0` to follow what's done in `add_entry`, see: <https://github.com/odoo/odoo/blob/8df1810ef08403a8c3bcc8310457359ad9640be8/odoo/tools/translate.py#L802>

Note: For code, the `res_id` correspond to the actual line number where the term appear on; so if the same term appears  multiple times in the same file (even with a `set`) we would write it multiple time.